handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -2025,7 +2022,7 @@ private static boolean isEndPointVerificationEnabled(String endPointIdentificati
     }
 
     private boolean isDestroyed() {
```

Comment:
are you sure all usages of this method are always in a synchronized block? I haven't done the investigation but The runnable created by `getDelegatedTask()` looks interesting. Also is it worth omitting the `synchronized` from the method signature to rely upon this not changing in the future? consider adding a comment if so (e.g. this must be called in a synchronized context)

Reply:
Yes I am sure... that said I forgot to make it volatile As @njhill pointed out 🤦‍♂️